builtInFunctions/esdtDataStorage.go

```diff
@@ -383,7 +383,16 @@ func (e *esdtDataStorage) saveESDTMetaDataToSystemAccount(
 	}
 
 	currentSaveData, _, err := systemAcc.AccountDataHandler().RetrieveValue(esdtNFTTokenKey)
-	if !mustUpdate && len(currentSaveData) > 0 {
+	if err != nil {
```

Comment:
this err was never used in the previous code.

Reply:
this will error only in case when the trie is NIL. So we might have an import DB issue for the first tx with system account.

Reply:
we will run an import-db for this. However, should have never happened